Nitrox.Launcher/Models/Extensions/ScreenExtensions.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Nitrox.Launcher.Models.Design;
+using Nitrox.Launcher.ViewModels;
+using Nitrox.Launcher.ViewModels.Abstract;
+
+namespace Nitrox.Launcher.Models.Extensions;
+
+public static class ScreenExtensions
+{
+    private static readonly List<RoutableViewModelBase> navigationStack = new();
+
+    /// <summary>
+    ///     Navigates to a view assigned to the given ViewModel.
+    /// </summary>
+    /// <param name="screen">The screen used to display the view.</param>
+    /// <param name="routableViewModel">ViewModel that should be shown.</param>
+    /// <typeparam name="TViewModel">Type of the ViewModel to show.</typeparam>
+    public static async Task ShowAsync<TViewModel>(this IRoutingScreen screen, TViewModel routableViewModel) where TViewModel : RoutableViewModelBase
+    {
+        if (screen == null)
+        {
+            return;
+        }
+        // When navigating away from a view in an async button command, busy states on buttons should also reset. Otherwise, when navigating back it would still show buttons being busy.
+        NitroxAttached.AsyncCommandButtonTagger.Clear();
+        if (screen.ActiveViewModel is RoutableViewModelBase routableViewModelBase)
+        {
+            navigationStack.RemoveAllFast(screen.ActiveViewModel, (item, param) => item.GetType() == param.GetType());
+            navigationStack.Add(routableViewModelBase);
+        }
+        Stopwatch sw = Stopwatch.StartNew();
+        Task contentLoadTask = routableViewModel.ViewContentLoadAsync();
+        // Only show loading screen if page isn't loading super quickly.
+        await Task.Delay(50);
+        if (!contentLoadTask.IsCompleted)
+        {
+            screen.ActiveViewModel = "Content is loading.. please wait";
```

Comment:
I'd rather have a WaitIndicator or a Spinner, the loading text is quite confusing.
Or maybe we could use some Shimmers view on heavy data loading